core/src/main/java/org/mule/execution/MessageProcessorNotificationExecutionInterceptor.java

```diff
@@ -120,7 +120,7 @@ public void processExceptionReplyTo(MessagingException exception, Object replyTo
         }
         finally
         {
-            if (!NonBlockingVoidMuleEvent.getInstance().equals(result) && fireNotification)
+            if (!(result instanceof VoidMuleEvent) && fireNotification)
```

Comment:
Why is this change required? Was not possible to receive a VoidMuleEvent before this PR?